### PR TITLE
Align with ARM, do not seperate PT_LOAD segment.

### PR DESCRIPTION
**Pull request by xyzhao2018**
Signed-off-by: Shirley Zhao <shirley.zhao@intel.com>

### CODE REVIEW

File: ta/os_test_lib/sub.mk [hunk 1]
```diff
@@ -1,6 +1,7 @@
 cppflags-$(WITH_TLS_TESTS) += -DWITH_TLS_TESTS=1
 global-incdirs-y += include
 srcs-y += os_test_lib.c
+LDFLAGS += -z noseparate-code
```
Comment (GangSecurity) [inline ta/os_test_lib/sub.mk:5]: do we need use #ifdef X86_64 here?

Comment (jingdlu) [inline ta/os_test_lib/sub.mk:5]: Do we need to use this flag for all of TAs?

Comment (xyzhao2018) [reply ta/os_test_lib/sub.mk:5]: done.

Comment (xyzhao2018) [reply ta/os_test_lib/sub.mk:5]: Both optee-os and all TAs use this flag. 
All align with ARM.

**Pull request closed without merging**